### PR TITLE
[WIP] Initialize some proxies for Fault Proof on Sepolia

### PR DESCRIPTION
**Pull request by cbfyi**
First part of rolling out Fault Proof on Sepolia for Base.

We need to initialize some contracts required for Fault Proof. They will be "owned" by the proxy admin owner / system admin. 

- `DisputeGameFactoryProxy`
- `DelayedWETHProxy`

`AnchorStateRegistryProxy` to follow in a separate set of changes once `DisputeGameFactoryProxy` is ready as the `AnchorStateRegistry` implementation depends on it.

These are prerequisites for Fault Proof for Base Sepolia.

The factory will not be useable just yet, until we `setImplementation` for specific game types. The game implementations will be deployed separately as they depend on the anchor state registry.

The proxies were deployed using our internal Smart Contract Manager (SCM). We will need the proxy admin to initialize these currently blank proxies.

### CODE REVIEW

File: sepolia/2024-06-04-initialize-fault-proof-proxies/Makefile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+include ../../Makefile
+include ../.env
+include .env
+
+ifndef LEDGER_ACCOUNT
+override LEDGER_ACCOUNT = 0
```
Comment (anikaraghu) [inline sepolia/2024-06-04-initialize-fault-proof-proxies/Makefile:6]: Most signers are using their second ledger address for testnet
```suggestion
override LEDGER_ACCOUNT = 1
```

File: sepolia/2024-06-04-initialize-fault-proof-proxies/script/InitializeFaultProofProxies.s.sol [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        console.log("DelayedWETH version: %s", weth.version());
+        require(weth.owner() == PROXY_ADMIN_OWNER);
+        // https://github.com/ethereum-optimism/optimism/blob/3aeb6bdd3efe798713098f890359513eaaa91522/packages/contracts-bedrock/deploy-config/mainnet.json#L52
+        require(weth.delay() == 604800);
+    }
+
+    /**
+     * @notice Creates the calldata
+     */
+    function _buildCalls() internal view override returns (IMulticall3.Call3[] memory) {
```
Comment (anikaraghu) [inline sepolia/2024-06-04-initialize-fault-proof-proxies/script/InitializeFaultProofProxies.s.sol:41]: Could we potentially link to Optimism's version of the script for reference?

File: sepolia/.env [hunk 3]
```diff
@@ -4,5 +4,6 @@ L2_RPC_URL=https://base-sepolia.cbhq.net
 L1_CHAIN_ID=11155111
 L2_CHAIN_ID=84532
 export PROXY_ADMIN=0x0389E59Aa0a41E4A413Ae70f0008e76CAA34b1F3
-export PROXY_ADMIN_OWNER=0x608081689Fe46936fB2fBDF7552CbB1D80ad4822
+export PROXY_ADMIN_OWNER=0x0fe884546476dDd290eC46318785046ef68a0BA9
```
[the remark this answers is not in the document]
Comment (anikaraghu) [reply sepolia/.env:7]: ah I thought we had updated, but let's definitely setup a PR if not

File: Makefile [hunk 4]
```diff
@@ -54,7 +54,8 @@ forge-deps:
 		github.com/OpenZeppelin/openzeppelin-contracts-upgradeable@v4.7.3 \
 		github.com/rari-capital/solmate@8f9b23f8838670afda0fd8983f2c41e8037ae6bc \
 		github.com/Saw-mon-and-Natalie/clones-with-immutable-args@105efee1b9127ed7f6fedf139e1fc796ce8791f2 \
-		github.com/Vectorized/solady@862a0afd3e66917f50e987e91886b9b90c4018a1
+		github.com/Vectorized/solady@cfb1568e5171c392a6d1150af26c8375b3d91831 \
```
Comment (anikaraghu) [inline Makefile:57]: curious if we ran into issues using the old version of solady? should probably update the template as well if so

File: sepolia/2024-06-04-initialize-fault-proof-proxies/.env [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# op-contracts/v1.4.0
+OP_COMMIT=547ea72d9849e13ce169fd31df0f9197651b3f86
+BASE_CONTRACTS_COMMIT=c14278a67d6eb029d3b9b505eea67d2b2ea637af
+
+# Fault Proof (proxies provisioned using SCM)
+DISPUTE_GAME_FACTORY_PROXY=0x2d05f4Fab724d5888c69C66A0f7FF21fFF7Db045
+DELAYED_WETH_PROXY=0xf74e8E7dCf6Ab718fB048ba5B69EFEF90c56999B
+
+# https://github.com/ethereum-optimism/superchain-registry/blob/46b9a85d619f88e006a43b641c00858920b47851/superchain/implementations/networks/sepolia.yaml#L23
+DISPUTE_GAME_FACTORY_IMPLEMENTATION=0xA51bea7E4d34206c0bCB04a776292F2f19F0BeEc
```
Comment (anikaraghu) [inline sepolia/2024-06-04-initialize-fault-proof-proxies/.env:10]: I see for this one there's 2 versions deployed, how do we know which to use?

File: sepolia/2024-06-04-initialize-fault-proof-proxies/script/InitializeFaultProofProxies.s.sol [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+            target: PROXY_ADMIN,
+            allowFailure: false,
+            callData: abi.encodeCall(
+                ProxyAdmin.upgradeAndCall,
+                (
+                    payable(DISPUTE_GAME_FACTORY_PROXY),
+                    DISPUTE_GAME_FACTORY_IMPLEMENTATION,
+                    abi.encodeCall(DisputeGameFactory.initialize, (PROXY_ADMIN_OWNER))
+                )
+            )
```
Comment (anikaraghu) [inline sepolia/2024-06-04-initialize-fault-proof-proxies/script/InitializeFaultProofProxies.s.sol:58]: do we also need to do these 2 calls? https://github.com/ethereum-optimism/optimism/blob/71b93116738ee98c9f8713b1a5dfe626ce06c1b2/packages/contracts-bedrock/scripts/fpac/FPACOPS.s.sol#L78-L79

File: sepolia/2024-06-04-initialize-fault-proof-proxies/.env [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# op-contracts/v1.4.0
+OP_COMMIT=547ea72d9849e13ce169fd31df0f9197651b3f86
```
Comment (anikaraghu) [inline sepolia/2024-06-04-initialize-fault-proof-proxies/.env:2]: confirmed, OP recommended using https://github.com/ethereum-optimism/optimism/releases/tag/op-contracts%2Fv1.4.0 which corresponds to this commit

**Review by anikaraghu — APPROVED** [review]

**Review by anikaraghu — APPROVED** [review]
lgtm!

**Comment by cb-heimdall** [comment]
<h1>✅ Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
✅
<code>1/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

**Pull request closed without merging**